Cannot find module 'genie-toolkit'

Run stanfordoval/almond-server with docker-compose. 
Error at startup:

> yarn run v1.22.5
$ node main.js
internal/modules/cjs/loader.js:638
    throw err;
Error: Cannot find module 'genie-toolkit'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/opt/almond/main.js:34:15)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12) 
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,30 +1,37 @@
-FROM docker.io/fedora:31
+FROM docker.io/fedora:33
 
 # Install all deps in the standard repos
 RUN dnf -y module install nodejs:10 && \
-    dnf -y install git curl pulseaudio-libs-devel unzip nodejs make gcc gcc-c++ libcanberra-devel atlas-devel blas-devel python2
-
-# Install yarn
-RUN curl -sL https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo
-RUN dnf -y install yarn
+    dnf -y install git curl pulseaudio-libs-devel zip unzip nodejs make \
+                   gcc gcc-c++ gettext libcanberra-devel atlas-devel blas-devel \
+                   python2 findutils
 
 RUN mkdir /opt/almond
-COPY . /opt/almond
+
+RUN useradd almond && \
+   chown almond:almond /opt/almond
+USER almond
+COPY --chown=almond:almond . /opt/almond
 WORKDIR /opt/almond
 
-# snowboy doesn't like building in docker, due to overlayfs bugs...
+# snowboy doesn't like building in docker, due to overlayfs bugs
 RUN rm -rf /opt/almond/node_modules && \
-    yarn && \
-    cp -Tr /usr/local/share/.cache/yarn/v6/npm-snowboy-1.3.1-220f23f026096fe5290d7919a9f0da93ccd253f2-integrity/node_modules/snowboy/ node_modules/snowboy/ && \
-    rm -fr /usr/local/share/.cache
-RUN cd node_modules/snowboy/ && \
-    yarn install --ignore-scripts --no-lockfile && \
-    PYTHON=python2 yarn run node-pre-gyp clean configure && \
+    npm ci && \
+    npm install --no-package-lock --ignore-scripts snowboy && \
+    cd node_modules/snowboy/ && \
+    PYTHON=python2 npx node-pre-gyp clean configure && \
     make -C build/ && \
-    rm -fr /usr/local/share/.cache
+    rm -fr /home/almond/.cache && \
+    rm -fr /home/almond/.npm
+
+RUN mkdir /home/almond/.cache
+
+ENV HOME /home/almond
+# switch back to root user so we can access the pulseaudio socker
+USER root
 
 EXPOSE 3000
 ENV THINGENGINE_HOME=/var/lib/almond-server
 ENV PULSE_SOCKET=unix:/run/pulse/native
 
-ENTRYPOINT [ "yarn", "start"]
+ENTRYPOINT ["npm", "start"]

--- a/Dockerfile.portable
+++ b/Dockerfile.portable
@@ -1,19 +1,18 @@
-FROM docker.io/fedora:30
+FROM docker.io/fedora:33
 
 # Install all deps in the standard repos
-RUN dnf -y install git unzip nodejs make gcc gcc-c++
-
-# Install yarn
-RUN curl -sL https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo
-RUN dnf -y install yarn
+RUN dnf -y install git zip unzip nodejs make gcc gcc-c++ gettext findutils
 
 RUN mkdir /opt/almond
-COPY . /opt/almond
-RUN rm -rf /opt/almond/node_modules
+
+RUN useradd almond && \
+   chown almond:almond /opt/almond
+USER almond
+COPY --chown=almond:almond . /opt/almond
 WORKDIR /opt/almond
-RUN yarn && rm -fr /root/.cache
+
+RUN npm ci && rm -fr /home/almond/.cache && rm -fr /home/almond/.npm
+RUN mkdir /home/almond/.cache
 
 EXPOSE 3000
-ENV THINGENGINE_HOME=/var/lib/almond-server
-
-ENTRYPOINT [ "yarn", "start"]
+ENTRYPOINT ["npm", "start"]

--- a/README.md
+++ b/README.md
@@ -19,9 +19,10 @@ the container to run as your regular user (and thus access PulseAudio from your 
 
 To run, use the command:
 ```bash
-podman run --name almond -p 3000:3000 --uidmap keep-id \
+podman run --name almond -p 3000:3000 \
     -v /dev/shm:/dev/shm \
     -v $XDG_RUNTIME_DIR/pulse:/run/pulse \
+    --security-opt label=disable \
     stanfordoval/almond-server
 ```
 

--- a/package.json
+++ b/package.json
@@ -35,7 +35,7 @@
     "sqlite3": "^5.0.0",
     "thingpedia": "~2.9.0-alpha.1",
     "uuid": "^8.3.1",
-    "ws": "^7.4.0"
+    "ws": "^7.4.2"
   },
   "optionalDependencies": {
     "canberra": "^0.1.2",
@@ -44,9 +44,9 @@
   },
   "devDependencies": {
     "coveralls": "^3.1.0",
-    "eslint": "^7.16.0",
+    "eslint": "^7.17.0",
     "form-data": "^3.0.0",
-    "nodemon": "^2.0.6",
+    "nodemon": "^2.0.7",
     "nyc": "^15.1.0",
     "parse5": "^6.0.1",
     "selenium-webdriver": "^4.0.0-alpha.8",

